src/compiler/checker/lines_32000_40000.rs

```diff
@@ -216,6 +216,20 @@ impl TypeChecker {
         self.check_source_element(Some(&*node.element_type));
     }
 
+    pub(super) fn check_union_or_intersection_type(
+        &mut self,
+        node: &Node, /*UnionOrIntersectionTypeNode*/
+    ) {
+        for_each(
+            node.as_union_or_intersection_type_node().types(),
```

Comment:
Per previous PR comment, I ran into a couple other places where it wanted to treat union + intersection types symmetrically (to grab their `types` field) so went ahead and created a `UnionOrIntersectionTypeNodeInterface` + `Node.as_union_or_intersection_type_node()` "unwrapper helper"